### DISCUSSION TITLE
Error while installing foundry on a mac

### DISCUSSION THREAD

**Discussion by Victoryeddy** [Cyfrin/foundry-full-course-cu#410] [Q&A]
I pasted this line in my terminal 
``curl -L https://foundry.paradigm.xyz | bash``
I got this 
 `` warning: libusb not found. You may need to install it manually on MacOS via Homebrew (brew install libusb).
After following the instruction to manually install libusb 
``
 brew install libusb
``
There was an error 
``
/usr/local/Homebrew/Library/Homebrew/standalone/sorbet.rb:4:in `require': cannot load such file -- sorbet-runtime (LoadError)
        from /usr/local/Homebrew/Library/Homebrew/standalone/sorbet.rb:4:in `<top (required)>'
        from /usr/local/Homebrew/Library/Homebrew/startup.rb:10:in `require_relative'
        from /usr/local/Homebrew/Library/Homebrew/startup.rb:10:in `<top (required)>'
        from /usr/local/Homebrew/Library/Homebrew/global.rb:4:in `require_relative'
        from /usr/local/Homebrew/Library/Homebrew/global.rb:4:in `<top (required)>'
        from /usr/local/Homebrew/Library/Homebrew/brew.rb:26:in `require_relative'
        from /usr/local/Homebrew/Library/Homebrew/brew.rb:26:in `<main>'``

**Comment by ChefAharoni** [comment]
Have you tried using ChatGPT to solve this issue?
This is what I got, try and update if you succeeded.
I would, however, avoid uninstalling homebrew, and try other fixes if the fixes before it didn't help.

```bash
brew update
brew upgrade
```

If that doesn't work, it's likely that something has gone wrong with the Homebrew installation itself, or there might be a problem with the Ruby environment that Homebrew relies on.

Since the issue seems to be with the `sorbet-runtime`, they could try reinstalling `sorbet`. To do this, they can follow these steps:

1. Uninstall `sorbet`:
    ```bash
    gem uninstall sorbet sorbet-runtime
    ```

2. Reinstall `sorbet`:
    ```bash
    gem install sorbet sorbet-runtime
    ```

If reinstalling `sorbet` doesn't solve the issue, it might be helpful to uninstall and reinstall Homebrew itself. Before they do this, remind your friend to take note of the packages they've installed via Homebrew, as they will need to reinstall them. 

To uninstall Homebrew, use this command:

```bash
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
```

To reinstall Homebrew, use this command:

```bash
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

After reinstalling Homebrew, your friend should try to install `libusb` again:

```bash
brew install libusb
```

If the issue persists even after these steps, it could be related to a deeper issue with your friend's Ruby or system environment, and additional troubleshooting may be required. 

Remember to always be careful when uninstalling and reinstalling system components, as this can sometimes lead to unintended consequences. Always back up important data before performing these actions.

**Reply by Victoryeddy** [reply]
Thank you . it has been resolved

**Reply by ChefAharoni** [reply]
Great, than please mark my previous answer as the answer, to let other people now it has been resolved.
